java/src/main/java/com/saucelabs/saucebindings/DeviceOrientation.java

```diff
@@ -0,0 +1,36 @@
+package com.saucelabs.saucebindings;
+
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public enum DeviceOrientation {
+    LANDSCAPE("landscape"),
+    PORTRAIT("portrait");
+
+    @Getter
+    private final String value;
+
+    private static final class DeviceOrientationLookup {
+        private static final Map<String, String> lookup = new HashMap<String, String>();
+    }
+
+    public static Set keys() {
+        return DeviceOrientationLookup.lookup.keySet();
```

Comment:
Could also be `return Stream.of(this.values()).map((x) -> x.name()).collect(toList())`

Reply:
Yeah, it took me a while to find an approach that worked for what I needed. This code matches what I'm doing in all the other enums. Is the an advantage to replacing the code in all of them?

Reply:
I would say it makes sense since we get rid of the redundant data structure. All operations could be done using native methods of the Enum class itself